Library/Formula/racer.rb

```diff
@@ -0,0 +1,72 @@
+class Racer < Formula
+  desc "Rust Code Completion utility"
+  homepage "https://github.com/phildawes/racer"
+
+  url "https://github.com/phildawes/racer/archive/v1.0.0.tar.gz"
+  sha256 "78895296ed688eeccbaf7745235f0fc503407bfa718f53583a4dcc9e1246b7f5"
+  head "https://github.com/phildawes/racer.git"
+
+  option "without-src", "Do not get rust source for racer from www.rust-lang.org"
+  option "with-beta-src", "Get beta version of rust source"
+  option "with-nightly-src", "Get nightly version of rust source"
+  depends_on "rust" => :build
+
+  resource "rustsrc" do
+    url "https://static.rust-lang.org/dist/rustc-1.3.0-src.tar.gz"
+    sha256 "ea02d7bc9e7de5b8be3fe6b37ea9b2bd823f9a532c8e4c47d02f37f24ffa3126"
+  end
+
+  resource "rustsrc-beta" do
+    url "https://static.rust-lang.org/dist/rustc-beta-src.tar.gz"
+    sha256 "e393758a541b1e1859c17deb54a548eb076ab9a7901210ddb47b033aa2d16e79"
+  end
+
+  resource "rustsrc-nightly" do
+    url "https://static.rust-lang.org/dist/rustc-nightly-src.tar.gz"
+    sha256 "be004a627cff79e7eb3ef17b2b855d6a5ee2258246c56cd6812b900ddf5ee79c"
+  end
+
+  def install
+    if build.with?("src")
```

Comment:
If you want to support different versions this is not the right way to do it. The same formula (racer) should work for most people. Let's see what options we have:

* stable: this is a must, we expect most people to use this
* best: some people use beta and in particular alongside with stable, so we must support both at the same time. It is being cut every week so you would want to update the formula quite often to make this work
* nightly: a lot of devs use nightly. But nightly is pushed every day so unless this formula is updated every day the sha won't match so the resource will be useless

Given the above my suggestion would be to support stable and at most beta. You should make them two different options so that people can install none, one, or both of them.